lingocore/lib/lesson_pages.dart

```diff
@@ -0,0 +1,240 @@
+
+  Question({required this.type, required this.questionText, this.options});
+
+  factory Question.fromJson(Map<String, dynamic> json) {
+    return Question(
+      type: json['type'],
+      questionText: json['question'],
+      options:
+          json['options'] != null ? List<String>.from(json['options']) : null,
+    );
+  }
+}
+
+class QuestionWidget extends StatelessWidget {
+  final Question question;
+  final Function(String userAns) checkAns;
+  const QuestionWidget({
+    super.key,
+    required this.question,
+    required this.checkAns,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    switch (question.type) {
+      case 'true_false':
+        return buildTrueFalse();
+      case 'multiple_choice':
+        return buildMultipleChoice();
+      default:
+        return const Text('bilinmeyen soru tipi');
+    }
+  }
+
+  Widget buildQuestionText(String text) {
+    return Text(question.questionText, style: const TextStyle(fontSize: 30));
+  }
+
+  // Generate button
+  Widget buildAnswerButton(String label, VoidCallback onPressed) {
+    return Padding(
+      padding: const EdgeInsets.all(15.0),
+      child: Container(
+        decoration: BoxDecoration(
+          gradient: LinearGradient(
+            colors: [Colors.greenAccent, Colors.white],
+            begin: Alignment.topLeft,
+            end: Alignment.bottomRight,
+          ),
+          borderRadius: BorderRadius.circular(10),
+        ),
+        child: ElevatedButton(
+          style: ElevatedButton.styleFrom(
+            padding: EdgeInsets.symmetric(vertical: 30, horizontal: 50),
+            elevation: 0,
+            shadowColor: Colors.transparent,
+            backgroundColor: Colors.transparent,
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(10),
+              side: BorderSide(color: Colors.green, width: 3),
+            ),
+          ),
+          onPressed: onPressed,
+          child: Text(label, style: TextStyle(fontSize: 30)),
+        ),
+      ),
+    );
+  }
+
+  // True/False Questions
+  Widget buildTrueFalse() {
+    return Column(
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(100.0),
+          child: buildQuestionText(question.questionText)
+        ),
+
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+          children: [
+            buildAnswerButton('Doğru', () => checkAns('Şık 1')),
+            buildAnswerButton('Yanlış', () => checkAns('Yanlış')),
+          ],
+        ),
+      ],
+    );
+  }
+
+  Widget buildMultipleChoice() {
+    return Column(
+      children: [
+        Padding(
+          padding: const EdgeInsets.all(50.0),
+          child: Text(
+            question.questionText,
+            style: const TextStyle(fontSize: 30),
+          ),
+        ),
+        ...(question.options ?? []).map(
+          (option) => buildAnswerButton(option, () => checkAns(option)),
```

Comment:
Ayrı bir column içine al veya grid şeklinde göster.
Okunaklılık için ... spread operatörünü kullanmak yerine map yapıp direkt children'a atayabilirsin. 
```suggestion
Column(
    children: question.options?.map(
      (option) => buildAnswerButton(
        option,
        () => checkAns(option)
      )
    ) ?? [],
)
```
Bunu değiştirince layout'u yeniden düzenlemen gerekebilir